### PR TITLE
Fix a number of Injection related issues

### PR DESCRIPTION
**Pull request by johnoliver**
- Fix packaging so that injection works live
- Add GraphQL tests

Unrelated fix:
- Don't apply version sanity checks to user supplied versions

# Checklist

- [ ] You added tests to cover the change
- [ ] `mvn clean install` build and test completes
- [ ] You changed or added to the documentation

### CODE REVIEW

File: adoptopenjdk-api-v3-persistence/src/main/kotlin/net/adoptopenjdk/api/v3/dataSources/APIDataStore.kt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (karianna) [inline adoptopenjdk-api-v3-persistence/src/main/kotlin/net/adoptopenjdk/api/v3/dataSources/APIDataStore.kt:1]: remove whitespace

File: adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/V3Updater.kt [hunk 2]
```diff
@@ -10,38 +10,39 @@ import net.adoptopenjdk.api.v3.dataSources.models.AdoptRepos
 import net.adoptopenjdk.api.v3.dataSources.persitence.ApiPersistence
 import net.adoptopenjdk.api.v3.models.Variants
 import net.adoptopenjdk.api.v3.stats.StatsInterface
-import org.jboss.weld.environment.se.Weld
 import org.slf4j.LoggerFactory
 import java.io.OutputStream
 import java.security.MessageDigest
 import java.util.*
```
Comment (karianna) [inline adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/V3Updater.kt:16]: java.util.*

File: adoptopenjdk-api-v3-updater/src/test/kotlin/net/adoptopenjdk/api/AdoptReposBuilderTest.kt [hunk 3]
```diff
@@ -1,117 +1,156 @@
...
+    fun updatedReleaseIsUpdatedWhenUpdated(repo: AdoptRepos, adoptReposBuilder: AdoptReposBuilder, stub: AdoptRepositoryStub) {
+        runBlocking {
+            assertTrue { repo.getFeatureRelease(8)!!.releases.getReleases().contains(stub.originaToUpdate) }
+            assertTrue { !updated!!.getFeatureRelease(8)!!.releases.getReleases().contains(stub.originaToUpdate) }
+            assertTrue { updated!!.getFeatureRelease(8)!!.releases.getReleases().contains(stub.toUpdate) }
             assertTrue { updated != repo }
         }
     }
 
+    /*
```
Comment (karianna) [inline adoptopenjdk-api-v3-updater/src/test/kotlin/net/adoptopenjdk/api/AdoptReposBuilderTest.kt:148]: remove commented code

File: adoptopenjdk-api-v3-updater/src/test/kotlin/net/adoptopenjdk/api/ReleaseVersionResolverTest.kt [hunk 4]
```diff
@@ -1,14 +1,9 @@
 package net.adoptopenjdk.api
 
 import kotlinx.coroutines.runBlocking
-import net.adoptopenjdk.api.v3.AdoptReposBuilder
 import net.adoptopenjdk.api.v3.JsonMapper
 import net.adoptopenjdk.api.v3.dataSources.*
```
Comment (karianna) [inline adoptopenjdk-api-v3-updater/src/test/kotlin/net/adoptopenjdk/api/ReleaseVersionResolverTest.kt:5]: Exapnd * import

**Review by karianna — CHANGES_REQUESTED** [review]
Nitpicks and a bit of a lingering question on if we've added DI why is there more code now? Question

**Comment by johnoliver** [comment]
@karianna the addition of code is mostly from:
- we are now testing GraphQl classes that were previously not testable so that added code. 
- we had not fully removed our dependency on having `example-data.json.gz` in the code base for tests. That has now been fully removed and we generate test data, again that added more code.
- A fair number of the additions are just xml files or annotations needed in place to configure the DI

**Review by karianna — APPROVED** [review]

**Comment by karianna** [comment]
@johnoliver - I'm happy to land if you fix the nitpicks :-)

**Comment by johnoliver** [comment]
@karianna done

**Pull request merged**